Refactor compose files into separate directory (#2753)
tenzir/vast: docker/compose/README.md

# Docker compose overlays

## Naming Compose files

All `.yaml` file in this directory being Docker Compose configuration files, we
ignore the convention of prefixing them with `docker-compose.`

Their naming follows the structure `{integration}.{feature}.yaml`:
- `integration`: the name of the core service that is being integrated
- `feature`:
  - if unspecified, the file is expected to contain the base configuration of
    the integration (it can usually work in standalone mode)
  - if `vast` is specified (e.g `quarto.vast.yaml`), it means that the file
    contains the configurations to interact with the `vast` service (e.g
    networking)
  - otherwise a keyword indicating what is being enabled in the `integration` on
    top of the base configuration (e.g `vast.volume.yaml` configures volumes for
    the `vast` service to provide persistence)

## The environment files

If a `.env` file is present at the base of the project directory (by default

> this directory), Docker Compose automatically loads environment from it. For

Reviewer: ```suggestion
this directory), Docker Compose automatically loads environment variables from it. For
```

each integration, we provide a `{integration}-env.example` file with the

> supported configurations (values are defaulted in the Docker Compose files as

Reviewer: ```suggestion
supported configuration options (values are defaulted in the Docker Compose files as
```

> well). You can optionnally create a `.env` file in the project directory and

Reviewer: ```suggestion
well). You can optionally create a `.env` file in the project directory and
```

> cherry pick the variables you want to customize from the example files.

Reviewer: ```suggestion
cherry-pick the variables you want to customize from the example files.
```

Reviewer: This last sentence is passive voice, might want to rephrase in terms of the action the user wants to take:

> To customize from the example files, create a `.env` file in the project directory and cherry-pick environment variables as needed.

Author: I feel it doesn't convey the fact that it's optional very well, does it?

Author: I propose

> Optionally, create a `.env` file in the project directory and cherry-pick the variables you want to customize from the example files.

Good for you?
